av1an-core/src/vapoursynth.rs

```diff
@@ -210,59 +213,105 @@ pub fn create_vs_file(
-      core.{}(r\"{}\", cachefile={:?}).set_output()",
+  // Only used for DGDECNV
+  let dgindex_path = match chunk_method {
+    ChunkMethod::DGDECNV => {
+      let dgindexnv_output = temp.join("split").join("index.dgi");
+
+      // Run dgindexnv to generate the .dgi index file
+      Command::new("dgindexnv")
+        .arg("-h")
+        .arg("-i")
+        .arg(&source)
+        .arg("-o")
+        .arg(&dgindexnv_output)
+        .output()?;
+
+      &to_absolute_path(&dgindexnv_output)?
+    }
+    _ => &source,
+  };
+
+  // Include rich loadscript.vpy and specify source, chunk_method, and cache_file
+  // Also specify downscale_height, pixel_format, and scaler for Scene Detection
+  // TODO should probably check if the syntax for rust strings and escaping utf and stuff like that is the same as in python
+  let mut load_script_text = include_str!("loadscript.vpy")
+    .replace(
+      "source = os.environ.get('AV1AN_SOURCE', None)",
+      &format!(
+        "source = r\"{}\"",
         match chunk_method {
-          ChunkMethod::FFMS2 => "ffms2.Source",
-          ChunkMethod::LSMASH => "lsmas.LWLibavSource",
-          _ => unreachable!(),
-        },
-        source.display(),
-        cache_file
-      )
-      .as_bytes(),
-    )?;
+          ChunkMethod::DGDECNV => dgindex_path.display(),
+          _ => source.display(),
+        }
+      ),
+    )
+    .replace(
+      "chunk_method = os.environ.get('AV1AN_CHUNK_METHOD', None)",
+      &format!("chunk_method = {chunk_method_lower:?}"),
+    )
+    .replace(
+      "cache_file = os.environ.get('AV1AN_CACHE_FILE', None)",
+      &format!("cache_file = {:?}", cache_file),
+    );
+
+  if let Some(scene_detection_downscale_height) = scene_detection_downscale_height {
+    load_script_text = load_script_text.replace(
+      "downscale_height = os.environ.get('AV1AN_DOWNSCALE_HEIGHT', None)",
+      &format!("downscale_height = {scene_detection_downscale_height}"),
+    );
   }
+  if let Some(scene_detection_pixel_format) = scene_detection_pixel_format {
+    load_script_text = load_script_text.replace(
+      "sc_pix_format = os.environ.get('AV1AN_PIXEL_FORMAT', None)",
+      &format!("pixel_format = \"{scene_detection_pixel_format:?}\""),
+    );
+  }
+  load_script_text = load_script_text.replace(
+    "scaler = os.environ.get('AV1AN_SCALER', None)",
+    &format!("scaler = {scene_detection_scaler:?}"),
+  );
+
+  load_script.write_all(load_script_text.as_bytes())?;
 
   Ok(load_script_path)
 }
 
+pub fn copy_vs_file(
+  temp: &str,
+  source: &Path,
+  downscale_height: Option<usize>,
+) -> anyhow::Result<PathBuf> {
+  let temp: &Path = temp.as_ref();
+  let scd_script_path = temp.join("split").join("scene_detection.vpy");
+  let mut scd_script = File::create(&scd_script_path)?;
+
+  let source_script = std::fs::read_to_string(source)?;
+  if let Some(downscale_height) = downscale_height {
+    let regex = Regex::new(r"(\w+).set_output\(").unwrap();
```

Comment:
Likewise, this regex and replacement may not be necessary at all if we can do it using the loadscript + env vars.

Reply:
Never mind, I see now why we're doing this here.